### DISCUSSION TITLE
Webviews not working

### DISCUSSION THREAD

**Discussion by jankais3r** [The-Wineskin-Project/WineskinServer#253] [Q&A]
Hi,

I have two very niche, industry-specific apps that utilize web views. However, the web views do not load.

**App 1:**
<img width="733" alt="relativity" src="https://user-images.githubusercontent.com/2159464/165567855-1bebdfbc-31d5-4353-8cf1-ff40910dabf5.png">
This app loads webpage that contains a login form in a pop-up window. But in wineskin the page is just blank. None of the context menu options do anything (e.g. view source).

**App 2 ([free download](https://go.exterro.com/l/43312/2022-01-21/f6h1s3)):**
<img width="1138" alt="ftk" src="https://user-images.githubusercontent.com/2159464/165568508-fe393481-9c21-4141-aa6a-648787ec8c95.png">
The second app has a bottom view that is used to display a preview of selected document - an image, PDF, or similar. However, when running under wineskin, it is again just white.

I tried installing ie8 package from winetricks, but that did not make any difference. Any idea how to get these webviews working? Besides this, all works great. Many thanks for building wineskin!

**Comment by jankais3r** [comment]
webview related logs from running App 2:
```
0024:fixme:ieframe:navigate_url Unsupported args (Flags 000000000021A188 {VT_INT: 14}; TargetFrameName (null))
0024:fixme:ieframe:ClientSite_GetContainer (00000000000E0698)->(000000000021F4F8)
0024:fixme:ieframe:DocHostUIHandler_GetDropTarget (00000000000E0698)
0024:fixme:ieframe:PropertyNotifySink_OnChanged unimplemented dispid 1005
0024:fixme:ieframe:navigate_url Unsupported args (Flags 000000000021D288 {VT_INT: 14}; TargetFrameName (null))
0024:fixme:ieframe:ClientSite_GetContainer (00000000000E0698)->(000000000021F4F8)
0024:fixme:ieframe:DocHostUIHandler_GetDropTarget (00000000000E0698)
0024:fixme:ieframe:PropertyNotifySink_OnChanged unimplemented dispid 1005
0024:fixme:ieframe:navigate_url Unsupported args (Flags 000000000021CE48 {VT_INT: 14}; TargetFrameName (null))
0024:fixme:ieframe:ClientSite_GetContainer (00000000000E0698)->(000000000021ED18)
0024:fixme:ieframe:DocHostUIHandler_GetDropTarget (00000000000E0698)
0024:fixme:ieframe:PropertyNotifySink_OnChanged unimplemented dispid 1005
0024:fixme:ieframe:navigate_url Unsupported args (Flags 000000000021F6C8 {VT_INT: 14}; TargetFrameName (null))
0024:fixme:mshtml:on_change_dlcontrol unsupported dlcontrol 40008790
0024:fixme:mshtml:nsChannelBSC_on_progress External MIME: L"application/octet-stream"
0024:fixme:mshtml:handle_extern_mime_navigation unimplemented in non-webbrowser mode
0024:fixme:ieframe:DocHostUIHandler_GetDropTarget (00000000000E0698)
0024:fixme:ieframe:navigate_url Unsupported args (Flags 000000000021CE48 {VT_INT: 14}; TargetFrameName (null))
0024:fixme:ieframe:DocHostUIHandler_GetDropTarget (00000000000E0698)
0024:fixme:ieframe:PropertyNotifySink_OnChanged unimplemented dispid 1005
0024:fixme:ieframe:ControlSite_OnFocus (00000000000E0698)->(1)
0024:fixme:ieframe:DocHostUIHandler_ShowContextMenu default action not implemented
0024:fixme:mshtml:OleCommandTarget_Exec unsupported nCmdID 28 of CGID_MSHTML group
0024:fixme:ieframe:DocHostUIHandler_ShowContextMenu default action not implemented
0024:fixme:mshtml:OleCommandTarget_Exec unsupported nCmdID 2283 of CGID_MSHTML group
0024:fixme:ieframe:DocHostUIHandler_ShowContextMenu default action not implemented
0024:fixme:ieframe:DocHostUIHandler_ShowContextMenu default action not implemented
0024:fixme:ieframe:DocHostUIHandler_ShowContextMenu default action not implemented
0024:fixme:mshtml:OleCommandTarget_Exec unsupported nCmdID 6042 of CGID_MSHTML group
0024:fixme:ieframe:DocHostUIHandler_ShowContextMenu default action not implemented
0024:fixme:mshtml:OleCommandTarget_Exec unsupported nCmdID 2139 of CGID_MSHTML group
0024:fixme:ieframe:OleInPlaceObject_InPlaceDeactivate (00000000000E0530)
0024:fixme:ieframe:ControlSite_OnFocus (00000000000E0698)->(0)
0024:fixme:ieframe:InPlaceSite_OnInPlaceDeactivateEx fNoRedraw (1) ignored
0024:fixme:mshtml:HlinkTarget_SetBrowseContext (00000000035AA730)->(0000000000000000)
```

I tried to install the latest [gecko package](https://dl.winehq.org/wine/wine-gecko/2.47.2/), but nothing changed.

**Reply by Gcenx** [reply]
I do remember seeing others having issues with application that use webview so it’s lightly an upstream wine problem.

Theres no need to manually install wine-gecko as it will request/download the .msi installers if needed. However the recent Engines and my upstream packages directly bundle wine-gecko so you’ll never be prompted to install as it will simply auto install.

**Reply by jankais3r** [reply]
Thanks for the reply, it indeed seems to be an upstream Wine issue, as there are [quite a few](https://bugs.winehq.org/buglist.cgi?quicksearch=ieframe%20navigate_url%20Unsupported%20args) very old yet unresolved bug reports…

**Reply by Gcenx** [reply]
I’m assuming this more being a winemac.drv problem as there’s known limitations, there’s some commits in wine-6.17 `winemac.drv: Enable layer-backed views.`

So that _might_ be in CX22 but there’s a possibility that might be reverted as it also broken anything below Mojave….

**Reply by Gcenx** [reply]
Here’s the mentioned regression related to the above commits https://bugs.winehq.org/show_bug.cgi?id=52354

**Reply by jankais3r** [reply]
Thanks. I will keep an eye on future engine updates and will report back.
